Source/WebCore/platform/network/DNS.cpp

```diff
@@ -76,4 +76,50 @@ std::optional<IPAddress> IPAddress::fromString(const String& string)
     return std::nullopt;
 }
 
+IPAddress IPAddress::isolatedCopy() const
+{
+    if (isIPv4())
+        return IPAddress { ipv4Address() };
+
+    if (isIPv6())
+        return IPAddress { ipv6Address() };
+
+    ASSERT_NOT_REACHED();
+    return IPAddress { WTF::HashTableEmptyValue };
+}
+
+unsigned IPAddress::matchingNetMaskLength(const IPAddress& other) const
+{
+    const unsigned char* addressData = nullptr;
+    const unsigned char* otherAddressData = nullptr;
+    size_t addressLengthInBytes = 0;
+    if (isIPv4() && other.isIPv4()) {
+        addressLengthInBytes = sizeof(struct in_addr);
+        addressData = reinterpret_cast<const unsigned char*>(&ipv4Address());
+        otherAddressData = reinterpret_cast<const unsigned char*>(&other.ipv4Address());
+    } else if (isIPv6() && other.isIPv6()) {
+        addressLengthInBytes = sizeof(struct in6_addr);
+        addressData = reinterpret_cast<const unsigned char*>(&ipv6Address());
+        otherAddressData = reinterpret_cast<const unsigned char*>(&other.ipv6Address());
+    }
+
+    if (!addressLengthInBytes)
```

Comment:
Maybe a comment on what this case means? Is it expected?

Reply:
Yes, it's expected in the case where the two `IPAddress`es are of different family (e.g. comparing v4 vs. v6) — I can see how this code looks confusing though, since it's not immediately obvious that this early return exists to catch that case.

I'll make this clearer by replacing this separate early return with an `else return;` above.